File: Assets/VirtualCity/ProtoDefine/Java/mall/RspGetGoodsMessage.java
package com.kingston.jforgame.server.game.mall.message;

import com.baidu.bjf.remoting.protobuf.FieldType;
import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;
import com.kingston.jforgame.server.game.Modules;
import com.kingston.jforgame.server.game.entity.mall.Goods;
import com.kingston.jforgame.server.game.mall.MallDataPool;
import com.kingston.jforgame.socket.annotation.MessageMeta;
import com.kingston.jforgame.socket.message.Message;

import java.util.List;

@MessageMeta(module = Modules.SHOPPING,cmd = MallDataPool.RSP_GETGOODS)
public class RspGetGoodsMessage extends Message {
    @Protobuf(order = 1)
    private int code;
    @Protobuf(order = 2)
    private String tip;

    /**商品List*/
    @Protobuf(order = 3,fieldType = FieldType.OBJECT)
    private List<Goods> goodsList;

    public RspGetGoodsMessage() {
    }

    public RspGetGoodsMessage(int code, String tip, List<Goods> goodsList) {
        this.code = code;
        this.tip = tip;
        this.goodsList = goodsList;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getTip() {
        return tip;
    }

    public void setTip(String tip) {
        this.tip = tip;
    }

    public List<Goods> getGoodsList() {
        return goodsList;
    }

    public void setGoodsList(List<Goods> goodsList) {
        this.goodsList = goodsList;
    }
}

File: Assets/VirtualCity/ProtoDefine/RspGetGoodsMessage.cs
using System.Collections.Generic;
using ProtoBuf;
namespace ProtoDefine {
[ProtoContract]



public class RspGetGoodsMessage {
[ProtoMember(1)]
    public int code;
[ProtoMember(2)]
    public string tip;

    /**商品List*/
[ProtoMember(3)]
    public List<Goods> goodsList;

    public RspGetGoodsMessage() {
    }

    public RspGetGoodsMessage(int code, string tip, List<Goods> goodsList) {
        this.code = code;
        this.tip = tip;
        this.goodsList = goodsList;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public string getTip() {
        return tip;
    }

    public void setTip(string tip) {
        this.tip = tip;
    }

    public List<Goods> getGoodsList() {
        return goodsList;
    }

    public void setGoodsList(List<Goods> goodsList) {
        this.goodsList = goodsList;
    }
}
}